language/movie_nl/main.py

```diff
@@ -0,0 +1,366 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import codecs
+import glob
+import json
+import logging
+import os
+
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+import httplib2
+from oauth2client.client import GoogleCredentials
+import requests
+import socket
+
+def analyze_document(service, document):
+    """Analyze the document and get the distribution of sentiments and
+    the movie name."""
+    logging.info('Analyzing {}'.format(document.doc_id))
+
+    sentences, entities = document.extract_all_sentences(service)
+
+    try:
+        sentiments = [get_sentiment(service, sentence) for sentence in sentences]
+    except HttpError as e:
+        raise e
+
+    return sentiments, entities
+
+
+def get_request_body(text, syntax=True, entities=True, sentiment=True):
+    """Creates the body of the request to the language api in
+    order to get an appropriate api response."""
+    body = {
+        'document': {
+            'type': 'PLAIN_TEXT',
+            'content': text,
+        },
+        'features': {
+            'extract_syntax': syntax,
+            'extract_entities': entities,
+            'extract_document_sentiment': sentiment,
+        },
+        'encoding_type': 'UTF32'
+    }
+
+    return body
+
+def get_response_with_retry(request, tries, retry=3):
+    """Get the response using re-try"""
+    try:
+        response = request.execute()
+        return response
+    except HttpError as e:
+        raise e
+    except socket.error as se:
+        if tries > retry:
+            raise se
+
+        logging.error('Re-trying the request {}'.format(tries))
+
+        tries+=1
+        return get_response_with_retry(request, tries, retry)
+
+
+def get_sentiment(service, sentence):
+    """Get the sentence-level sentiment."""
+    body = get_request_body(
+        sentence, syntax=False, entities=True, sentiment=True)
+
+    docs = service.documents()
+    request = docs.annotateText(body=body)
+
+    response = get_response_with_retry(request, 1)
+
+    sentiment = response.get("documentSentiment")
+
+    if sentiment is None:
+        return (None, None)
+    else:
+        pol = sentiment.get("polarity")
+        mag = sentiment.get("magnitude")
+
+    if pol is None and mag is not None:
+        pol = 0
+    return (pol, mag)
+
+
+class Document(object):
+    """Document class captures a single document of movie reviews."""
+
+    def __init__(self, text, doc_id, doc_path):
+        self.text = text
+        self.doc_id = doc_id
+        self.doc_path = doc_path
+        self.sentent_pair = None
+        self.label = None
+
+    def extract_all_sentences(self, service):
+        """Extract the sentences in a document."""
+
+        if self.sentent_pair is None:
```

Comment:
nit: I'd just check if it's `not None` and return it immediately, which lets you dedent the rest of the function.